File: app/src/main/java/seucaioo/com/br/testeretrofit/ui/MainActivity.java
package seucaioo.com.br.testeretrofit.ui;

import android.os.Bundle;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.widget.Toast;

import java.util.ArrayList;
import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import seucaioo.com.br.testeretrofit.R;
import seucaioo.com.br.testeretrofit.retrofit.RetrofitInitializer;
import seucaioo.com.br.testeretrofit.retrofit.DataService;
import seucaioo.com.br.testeretrofit.model.Data;
import seucaioo.com.br.testeretrofit.model.DataResponse;

public class MainActivity extends AppCompatActivity {
    private List<Data> dataList;
    RecyclerView recyclerView;
    SwipeRefreshLayout swipeRefreshLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();
    }

    private void initViews() {
        dataList = new ArrayList<>();

        recyclerView = findViewById(R.id.recyclerView);
        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(this);
        recyclerView.setLayoutManager(mLayoutManager);
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setAdapter(new DataAdapter(this,dataList));

        swipeRefreshLayout = findViewById(R.id.swipeRefreshLayout);
        swipeRefreshLayout.setOnRefreshListener(OnRefreshListener());

        loadJSON();
    }

    private SwipeRefreshLayout.OnRefreshListener OnRefreshListener() {
        return new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                loadJSON();
            }
        };
    }

    private void loadJSON() {
        try {
            RetrofitInitializer retrofit = new RetrofitInitializer();
            DataService service = retrofit.dataService();

            Call<DataResponse> call = service.getData();
            call.enqueue(new Callback<DataResponse>() {
                @Override
                public void onResponse(Call<DataResponse> call, Response<DataResponse> response) {
                    if (response.isSuccessful()) {
                        swipeRefreshLayout.setRefreshing(false);
                        List<Data> dataList = response.body().getData();
                        if (dataList != null) {
                            for (Data d : dataList) {
                                recyclerView.setAdapter(
                                        new DataAdapter(getApplicationContext(), dataList));
                                Log.d("App", String.format(
                                        "ID: %s -> Name: %s -> Pwd: %s", d.getId(), d.getName(), d.getPwd()));
                            }
                        } else {
                            Log.d("App", "Erro: " + String.valueOf(response.code()));
                        }
                    }
                }

                @Override
                public void onFailure(Call<DataResponse> call, Throwable t) {
                    swipeRefreshLayout.setRefreshing(false);
                    recyclerView.setAdapter(new DataAdapter(getApplicationContext(), dataList));
                    Log.d("App", t.getMessage());
                    Toast.makeText(MainActivity.this,
                            "Erro ao buscar dados.\nVerifique sua conexão com a Internet !",
                            Toast.LENGTH_LONG).show();
                }
            });
        } catch (Exception e) {
            Log.d("App", e.getMessage());
            Toast.makeText(this, "Erro ", Toast.LENGTH_LONG).show();
        }
    }

}

File: app/src/main/java/seucaioo/com/br/testeretrofitkotlin/ui/MainActivity.kt
package seucaioo.com.br.testeretrofitkotlin.ui

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v4.widget.SwipeRefreshLayout
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.LinearLayoutManager
import android.util.Log
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import seucaioo.com.br.testeretrofitkotlin.R
import seucaioo.com.br.testeretrofitkotlin.model.Data
import seucaioo.com.br.testeretrofitkotlin.retrofit.RetrofitInitializer
import seucaioo.com.br.testeretrofitkotlin.model.DatasResponse

class MainActivity : AppCompatActivity() {

    val dataList = listOf<Data>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        init()
    }

    private fun init() {
        val recyclerView = recyclerView
        val mLayoutManager = LinearLayoutManager(this)
        recyclerView.layoutManager = mLayoutManager
        recyclerView.itemAnimator = DefaultItemAnimator()
        recyclerView.adapter = DataAdapter(this,dataList)

        val swipeRefreshLayout = swipeRefreshLayout
        swipeRefreshLayout.setOnRefreshListener(OnRefreshLayout())

        loadJSON()
    }

    private fun OnRefreshLayout(): SwipeRefreshLayout.OnRefreshListener {

        return SwipeRefreshLayout.OnRefreshListener { loadJSON() }
    }

    private fun loadJSON() = try {

        val retrofit = RetrofitInitializer()
        val service = retrofit.dataService()

        val call : Call<DatasResponse> = service.getDatas()
        call.enqueue( object : Callback<DatasResponse?> {
            override fun onResponse(call: Call<DatasResponse?>?, response: Response<DatasResponse?>?) {
                response?.let {
                    swipeRefreshLayout.isRefreshing = false
                    response?.body()?.datas.let {
                        val dataList : List<Data>? = it
                        if (dataList != null) {
                            for (d in dataList) {
                                recyclerView.adapter = DataAdapter(applicationContext, dataList)
                                Log.d("App", String.format("%s: %s -> %s", d.name, d.pwd, d.id))
                            }
                        } else {
                            Toast.makeText(baseContext,
                                    "Falha: " + response.code().toString(),
                                    Toast.LENGTH_LONG).show()
                        }
                    }
                }
            }

            override fun onFailure(call: Call<DatasResponse?>?, t: Throwable?) {
                swipeRefreshLayout.isRefreshing = false
                recyclerView.adapter = DataAdapter(applicationContext, dataList)
                Log.d("App", t?.message)
                Toast.makeText(this@MainActivity, "Erro ao buscar Dados!",
                        Toast.LENGTH_SHORT).show()
            }
        }
        )

/*
        call.enqueue(object : Callback<DatasResponse> {
            override fun onResponse(call: Call<DatasResponse>, response: Response<DatasResponse>) {
                if (response.isSuccessful()) {
                    swipeRefreshLayout.isRefreshing = false
                    val dataList = response.body()!!.datas
                    if (dataList != null) {
                        for (d in dataList!!) {
                            recyclerView.adapter = DataAdapter(applicationContext, dataList)
                            Log.d("App", String.format(
                                    "%s: %s -> %s", d.name, d.pwd, d.id))
                        }
                    } else {
                        Toast.makeText(baseContext,
                                "Falha: " + response.code().toString(),
                                Toast.LENGTH_LONG).show()
                    }

                }

            }

            override fun onFailure(call: Call<DatasResponse>, t: Throwable) {
                swipeRefreshLayout.isRefreshing = false
                val dataList = listOf<Data>()
                recyclerView.adapter = DataAdapter(applicationContext, dataList)
                Log.d("App", t.message)
                Toast.makeText(this@MainActivity, "Erro ao buscar Dados!",
                        Toast.LENGTH_SHORT).show()
            }
        })

*/

    } catch (e: Exception) {
        Log.d("App", e.message)
        Toast.makeText(this, e.toString(), Toast.LENGTH_LONG).show()
    }



}
